### ISSUE TITLE
Mobile - Select organization on registration

### ISSUE THREAD

**Issue by palisadoes** [PalisadoesFoundation/talawa#2330]
**Is your feature request related to a problem? Please describe.**

We are working on migrating the approval of requests to join organizations to a new Admin screen as outlined in this issue:

- https://github.com/PalisadoesFoundation/talawa-admin/issues/1416

Prior to this the process was:

1. New registrants joined the Community
1. Super Admins would approve the request
1. Approved registrants would login and select an organization

There were disadvantages to this approach:

1. This gave too much responsibility to the Super Admin who may not necessarily know people requesting to join in the way an Admin would. 
1. The two step process where the organization was selected post approval, could have been reduced to a simpler single step.

**Describe the solution you'd like**

_TALAWA API_

- This new requirement means that:
    1.  New `USER`s and `NON_USER`s will need to select an organization when registering so that the request can be directed to the correct organization's Admin screens.
    1. The setup script needs to create a single organization with a generic name on installation to facilitate registration to choice of at least one organization
    1. There can never be less than one organization in the DB
    1. The registration mutation must include organization ID information that issue https://github.com/PalisadoesFoundation/talawa-admin/issues/1416 will be able to detect.
    1. The current GitHub action pull request workflow needs to be updated so that the default organization is created successfully. This is in addition to the already existing test data importation test.
    1. The default organization creation mode mutation needs set the registration approval required flag to True to err on the side of caution.

_TALAWA MOBILE_
- This new requirement means that:
    1.  The user registration screen(s) will need to prompt for the organization and use the updated `Talawa API` registration mutation.
    1. The screen(s) design must follow the guidelines of our Design System found here:
        - https://docs.talawa.io/docs/design/ux/ux-design-system

_TALAWA ADMIN_
- This new requirement means that:
    1.  The user registration screen(s) will need to prompt for the organization and use the updated `Talawa API` registration mutation.
    1. The screen(s) design must follow the guidelines of our Design System found here:
        - https://docs.talawa.io/docs/design/ux/ux-design-system

_OTHER_
1. You will need to write tests for the updated code 
1. All other functionality must be maintained

**Describe alternatives you've considered**

- N/A

**Approach to be followed (optional)**

- See above

**Additional context**
1. Related issues:
     1. https://github.com/PalisadoesFoundation/talawa-api/issues/1703
     2. x
     3. x 
1. User definitions in the DB schema
![image](https://github.com/PalisadoesFoundation/talawa-api/assets/16875803/4b8cb6bb-7b38-4dde-ad8d-136bb0877349)

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

**Comment by Dante291** [comment]
@palisadoes I would like to work on this, please assign me

**Comment by Azad99-9** [comment]
@noman2002 @Cioppolo14 @Kevoniat could you please assign this issue to me.

**Comment by AVtheking** [comment]
I would like to work on this issue please assign me this issue

**Comment by abhisheksharma010** [comment]
Hello, can you assign this issue to me? I can easily resolve this.

**Comment by palisadoes** [comment]
Removing issue assignment. No activity. No PR.

**Comment by abhisheksharma010** [comment]
can you assign this issue to me?

**Comment by Dante291** [comment]
@palisadoes @Cioppolo14  this need to be fixed from backend side first only then i could work on this, waiting for this [PR](https://github.com/PalisadoesFoundation/talawa-api/pull/1741) to get merged, please assign me again.

**Comment by palisadoes** [comment]
@Dante291 

1. The blocking API PR for this issue has been merged. 
    1. https://github.com/PalisadoesFoundation/talawa-api/pull/1741
1. Please proceed with your work on this with urgency

**Comment by Dante291** [comment]
@palisadoes since you reverted the PR, I am going to wait for it to get merged again and then will start working on this

**Comment by Cioppolo14** [comment]
@Dante291 Are you working on this?

**Comment by Dante291** [comment]
@Cioppolo14 Yes, this need to be fixed from backend side first only then i will be able to work on this though.

**Comment by palisadoes** [comment]
1. Be aware of this too. The name and functionality of a mutation will soon change
    1. https://github.com/PalisadoesFoundation/talawa-api/issues/1859#issuecomment-1949567673

**Comment by Azad99-9** [comment]
@Cioppolo14 if this issue is vacant, I would like to work on this. Please assign me.

**Comment by Azad99-9** [comment]
@palisadoes please clarify, will the generic organisation always have `userRegistrationRequired` field set to true. So that the users can directly register into the app.

**Comment by Azad99-9** [comment]
@palisadoes please clarify, will the generic organisation always have userRegistrationRequired field set to false. So that the users can directly register into the app.

**Comment by palisadoes** [comment]
You cannot assume this. The Admin team may change that at any time after installation.

**Comment by Cioppolo14** [comment]
Unassigning due to no activity or open PR.

**Comment by Azad99-9** [comment]
@Cioppolo14 this issue has a backend dependency. Please have a look, the PR blocking this issue is https://github.com/PalisadoesFoundation/talawa-api/pull/1770

**Comment by Cioppolo14** [comment]
@Azad99-9 We are working on getting that one merged.

**Comment by palisadoes** [comment]
@Azad99-9 

The API PR is now complete. You can proceed, if you are available.

https://github.com/PalisadoesFoundation/talawa-api/issues/1703

**Comment by Azad99-9** [comment]
Sure @palisadoes ,@Cioppolo14 please assign me this issue.

**Comment by Azad99-9** [comment]
The backend bug needs to be fixed to proceed with this issue.

**Comment by Azad99-9** [comment]
https://github.com/PalisadoesFoundation/talawa-api/issues/2242

There is a prerequisite backend issue raised for this issue.

**Comment by gurramkarthiknetha** [comment]
if no one is currently working on this issue, I would like to take it on. Please assign it to me

**Comment by gurramkarthiknetha** [comment]
> @gurramkarthiknetha Our policy is to assign no more than two issues to each contributor across all repositories. This way everyone gets a chance to participate in the projects. We sometimes give exceptions for more urgent cases and sometimes we lose track, but the policy stands. You have reached your limit, please wait until your existing issues are closed before requesting more issues. You could unassign yourself from one of the other issues too.

could you assign this issue now ??

**Comment by MuhammadZeeshanAkram** [comment]
@palisadoes 
@Dante291 
@Cioppolo14 
@Azad99-9 
Hi, the backend is running without errors but I am facing issue during signup process as there are no organizations appearing.

https://github.com/user-attachments/assets/c3a92f3c-a362-40ba-93b6-8ba9a09a56cd

**Comment by palisadoes** [comment]
Please work with @rahulch07

**Comment by Cioppolo14** [comment]
Unassigning due to inactivity and/or no submitted PR. Others need to get a chance. Please let us know whether you want to be reassigned before someone else requests assignment or select another issue.

**Comment by Dante291** [comment]
@MohitMaulekhi your PR resolves this right?

**Comment by MohitMaulekhi** [comment]
> [@MohitMaulekhi](https://github.com/MohitMaulekhi) your PR resolves this right?

Yes

**Comment by BHARATH0153** [comment]
@palisadoes can you please assign me

**Comment by BHARATH0153** [comment]
@palisadoes please assign me

**Comment by yeswanthsai0718-cpu** [comment]
@palisadoes @Azad99-9 can i work on this

**Comment by MohitMaulekhi** [comment]
@palisadoes this issue needs to be closed as selecting an organisation while registering has already been implemented
